source/hud/blocklyButton.js

```diff
@@ -22,8 +22,16 @@ this.onClick = function() {
     var hud = this.parent;
     hud.lastBlocklyNodeID = hud.lastBlocklyNodeID || this.scene.player.rover.id;
     if ( hud.lastBlocklyNodeID !== undefined ) {
+        
+        var blocklyNode = blocklyNodes[ hud.lastBlocklyNodeID ];
+
+        if ( blocklyNode && Blockly.mainWorkspace ) {
+            Blockly.mainWorkspace.maxBlocks = blocklyNode.ramMax;
+        } else {
+            console.log("Either blocklyNode or Blockly.mainWorkspace are missing!");
+        }
```

Comment:
I actually injected that test printout above Line 326, in the switch statement around [Line 293](https://github.com/virtual-world-framework/mars-game/blob/ram-grayout-fix/source/view/index.js#L293). 

I thought that it would have been caught before Line 326, but it isn't because `blocklyNode` always seems to be undefined when `satProperty` is called in `index.js`. Currently investigating...